pipeline_api/src/main/java/com/splicemachine/pipeline/client/BulkWriteAction.java

```diff
@@ -465,12 +473,48 @@ private Exception parseIntoException(BulkWriteResult response){
         return first;
     }
 
+    /**
+     *
+     * s = submit time
+     * r = response time
+     * rc = refresh cache finish time.
+     * b = blocked time in synchronized method
+     *
+     * The algorithm will not immediately refresh for any calls with submit time (s) < rc
+     *
+
+     * |
+     * |
+     * |   s----------------r-bbs------r              (Blocked in synchronized block, until rc and then s < rc)
+     * |
+     * |         s----------------r s---r        (Ignore since s < rc
+     * |
+     * |      s----------------r    s---r        (Ignore since s < rc
+     * |
+     * |        s---------r---rc            (Refresh)
+     * |
+     * |                        s---------r (This would be allowed to invalidate cache if it is incorrect)...
+     * |
+     * ----------------------------------------------------------------------------------> t
+     *
+     *
+     * @param writerFactory
+     * @param tableName
+     */
+    public static synchronized void invalidateCache(BulkWriterFactory writerFactory, byte[] tableName, long submitTime) throws IOException {
```

Comment:
If we have multiple tables that need refreshing, we would serialize their invalidations because the latter ones would be blocked during the invalidation of the first one. This might not happen to frequently from a SparkExecutor but it might be common when forwarding index writes to several indexes. Could we make sure a thread only blocks if there's already an invalidation going on for the same table?